### PR TITLE
T3T1 - additional wallet backup info

### PR DESCRIPTION
**Pull request by obrusvit**
This PR adds:
- info about repeated words in WalletCreation flow,
- ability to display page counter information in `Footer`, e.g. "1 / 20".
- alternative styles for numerals of TT Satoshi font
- translations update

TODO:
- [x] update new strings in other langs
- [x] center the whole PageCounter content (not just around the foreslash)

### CODE REVIEW

File: core/embed/rust/src/ui/model_mercury/component/footer.rs [hunk 1]
```diff
@@ -224,3 +258,77 @@ impl crate::trace::Trace for Footer<'_> {
...
+        };
+
+        let foreslash_offset = Offset::x(10); // half of foreslash icon (6px) + spacing (4px)
+        let num_base_y = self.font.vert_center(self.area.y0, self.area.y1, "1");
+        let base_num = Point::new(self.area.center().x, num_base_y);
+        let base_num_curr = base_num - foreslash_offset;
+        let base_num_max = base_num + foreslash_offset;
+
+        let string_curr = build_string!(10, inttostr!(self.page_curr));
+        let string_max = build_string!(10, inttostr!(self.page_max));
```
Comment (obrusvit) [inline core/embed/rust/src/ui/model_mercury/component/footer.rs:317]: This should use the `uformat!` macro once merged.

File: core/embed/rust/src/ui/model_mercury/component/footer.rs [hunk 2]
```diff
@@ -2,27 +2,32 @@ use crate::{
...
+/// A host of this component is responsible of providing the exact area
+/// considering also the spacing. The height must be 18px (only instruction) or
+/// 37px (instruction and description/position). The content and style of both
+/// is configurable separatedly.
 #[derive(Clone)]
 pub struct Footer<'a> {
     area: Rect,
     text_instruction: TString<'a>,
     text_description: Option<TString<'a>>,
+    counter: Option<PageCounter>,
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/footer.rs:30]: Nit: for the sake of structure size this should probably be an enum sharing space with `text_description`, `style_description` since both can't be used at once IIUC.

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs [hunk 3]
```diff
@@ -18,11 +21,117 @@ use heapless::{String, Vec};
...
+type IndexVec = Vec<usize, MAX_WORDS>;
 
 /// Component showing mnemonic/share words during backup procedure. Model T3T1
 /// contains one word per screen. A user is instructed to swipe up/down to see
 /// next/previous word.
+/// This is a wrapper around a Frame so that the subtitle and Footer of the
+/// Frame can be updated based on the index of the word shown. Actual share
+/// words are rendered within `ShareWordsInner` component,
 pub struct ShareWords<'a> {
+    area: Rect,
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/share_words.rs:33]: Doesn't seem to be used anywhere, same for ShareWordsInner.

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs [hunk 4]
```diff
@@ -18,11 +21,117 @@ use heapless::{String, Vec};
 
 const MAX_WORDS: usize = 33; // super-shamir has 33 words, all other have less
 const ANIMATION_DURATION_MS: Duration = Duration::from_millis(166);
+type IndexVec = Vec<usize, MAX_WORDS>;
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/share_words.rs:24]:
```suggestion
type IndexVec = Vec<u8, MAX_WORDS>;
```
I'm a fan of using correct int types but `usize` seems wasteful in such large array. Other possibility is to use some kind of bitvector.

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs [hunk 5]
```diff
@@ -190,23 +293,67 @@ impl<'a> Component for ShareWords<'a> {
...
+        ];
+        let words4 = [
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+        ];
+
+        assert_eq!(ShareWords::find_repeated(&words1), IndexVec::new());
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/share_words.rs:345]: Hyper nit: tempted to add a test for empty input array.

**Review by mmilata — COMMENTED** [review]
Was hoping that we could implement a variant of Frame that holds any content which can provide current page via some (new) trait. But I suppose it's not as straightforward and am OK with the current approach. Bonus points for unit test.

File: core/embed/rust/src/ui/model_mercury/component/footer.rs (hunk 2 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/footer.rs]: reworked here https://github.com/trezor/trezor-firmware/pull/3954/commits/94305399464081fb22fe8efe170bf76e09a8149a 

I also removed `style_description` and `style_instruction` (and their setters) and made them `const` as they are consistent across all designs. The only reason why it was there was to support `PageCounter` color (green when max, e.g. `20 / 20`). But that's now handled internally by `PageCounter` component.

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs (hunk 3 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/share_words.rs]: Thanks, removed everywhere https://github.com/trezor/trezor-firmware/pull/3954/commits/856b4578701228e136642e379a8e00c2de9aa2a7

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs (hunk 4 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/share_words.rs]: https://github.com/trezor/trezor-firmware/pull/3954/commits/94c4b7ff7b39a2bf103008c08f60248d17044fee

File: core/embed/rust/src/ui/model_mercury/component/share_words.rs (hunk 5 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/share_words.rs:341]: https://github.com/trezor/trezor-firmware/pull/3954/commits/94c4b7ff7b39a2bf103008c08f60248d17044fee

**Review by mmilata — APPROVED** [review]
let's rebase & merge when CI passes

**Comment by bosomt** [comment]
**QA OK**

- info about repeated word is visible, very nice !
- footer x/20 is present

**Info:**
 - Suite version: desktop 24.6.2 (b319fb1a1e4e2fb85e398589dec064c26450bab0)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.6.2 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.0 regular (revision dd4671a5104952ef505d28d1f9e94d1484b4607a)
 - Transport: BridgeTransport 2.0.33
